test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -548,4 +555,43 @@ UCS_TEST_P(test_uct_cm_sockaddr, cm_server_reject)
                 (TEST_CM_STATE_SERVER_CONNECTED | TEST_CM_STATE_CLIENT_CONNECTED)));
 }
 
+UCS_TEST_P(test_uct_cm_sockaddr, server_switch_cm)
+{
+    UCS_TEST_MESSAGE << "Testing " << m_listen_addr
+                     << " Interface: " << GetParam()->dev_name;
+
+    m_switch_cm = true;
+
+    ucs_async_context_t *async;
+    uct_worker_h worker_test;
+    ucs_status_t status;
+
+    status = ucs_async_context_create(UCS_ASYNC_MODE_THREAD_SPINLOCK, &async);
+    ASSERT_UCS_OK(status);
+
+    status = uct_worker_create(async, UCS_THREAD_MODE_SINGLE, &worker_test);
+    ASSERT_UCS_OK(status);
+
+    status = uct_cm_open(GetParam()->component, worker_test, &m_cm_test);
+    ASSERT_UCS_OK(status);
+
+    cm_listen_and_connect();
+
+    wait_for_bits(&m_cm_state, TEST_CM_STATE_SERVER_CONNECTED |
+                               TEST_CM_STATE_CLIENT_CONNECTED);
+    EXPECT_TRUE(ucs_test_all_flags(m_cm_state, (TEST_CM_STATE_SERVER_CONNECTED |
+                                                TEST_CM_STATE_CLIENT_CONNECTED)));
+
+    cm_disconnect(m_client);
+
+    /* destroy the server's ep here (and not in the test's destroy flow) since
+     * the cm it is using was created in this test and therefore needs to be
```

Comment:
do you mean that we must destroy the new ep before destroying the new cm?

Reply:
yes. same as we destroy eps before destroying their cms (or ifaces), workers, context..